[RFC] Issue #9090 Update calls to '[en|de]codeURICompoonent()' functions

yybalam opened the pull request:
So, this PR change the `espace()` and `unescape()` calls of those deprecated functions to recommended `encodeURIComponent()` and `decodeURIComponent()` respectively.

Also I added a `testEncoding()` method for compare results. A string of special characters looks work fine `'~!@#$%^&*(){}[]=:/,;?+\'"\\'`, but some unicode does not `'äëíöü'`.

So, let's open the discussion. :sweat_smile:

yybalam committed:
Fix #9090
Call to 'encodeURI' and 'decodeURI' instead of 'escape' and 'unescape' deprecated functions.

yybalam committed:
Merge remote-tracking branch 'upstream/master'

yybalam committed:
Changed calls to '[en|de]codeURIComponent()' instead of '[en|de]codeURI()' function.
This change works fine for special characters ('~!@#$%^&*(){}[]=:/,;?+\'"\\') escaped by deprecated 'escape()' and 'unescape()' functions,
but fail for unicode characters ('äëíöü').

yybalam committed:
Added test for compare behavior between 'escape()' and 'unescape()' deprecated functions with recommended '[en|de]codeURIComponent()' functions by Cookie handler.

johnspackman commented:
Please can you add some documentation about backwards compatibility?

oetiker commented:
since `decodeURIComponent(escape('ä'))` actually raises an exception, how about wrapping the `decodeURIComponent` call into a `try` and in the `catch` branch handle the `URIError` condition by returning `null`

yybalam committed:
Added try/catch block for 'decodeURIComponent()' call.

yybalam committed:
Added test for assert null when 'decodeURIComponent()' throw an exception.

yybalam committed:
Added some changelog documentation.

yybalam commented:
Done. I'm not sure if my english and explanation are good enough. :sweat_smile:

oetiker commented:
@yybalam I think @johnspackman meant a comment in the jsdoc at the top of the two methods affected by the change

johnspackman commented:
yes 👍  that comment in changelog just needs to go where the API viewer will pick it up.

Your english is fine, although I think instead of "There no must be issues" you mean "There are no issues"

cajus commented on framework/source/class/qx/bom/Cookie.js:
```diff
@@ -62,7 +62,13 @@ qx.Bootstrap.define("qx.bom.Cookie",
+        return decodeURIComponent(document.cookie.substring(len, end));
+      }
+      catch (URIError) {
+        console.error(URIError.message);
```
That should be something like:

`qx.log.Logger.error("Error decoding URI components", URIError.message);`

yybalam committed:
Now when URIError is cached is also registred at qx logger.

level420 commented:
:ballot_box_with_check: CALL FOR VOTES

oetiker commented:
@yybalam please add the entry to the Changelog to the class documentation so that it is visible to people looking at the api viewer. The english is fine

yybalam committed:
Added documentation note to class for be avaible by the API viewer.
Also done a little change about special characters, there is no issues with them.

level420 commented:
@oetiker should we restart with a new voting or are we good to go?

oetiker commented:
@level420 I guess we are ok :) ... I would have mentioned that it was versions before qx 6.0 ... but alas

level420 commented:
5 Votes. Merging.
Thank you @yybalam
